ui/src/features/freight-timeline/open-container-initiative.ts

```diff
@@ -0,0 +1,71 @@
+import { formatDistance } from 'date-fns';
+
+const ociPrefix = 'org.opencontainers.image';
+
+type Annotation = Record<string, string>;
+
+export const ociAnnotationKeys = {
+  // date and time on which the image was built
+  createdAt: `${ociPrefix}.created`,
+  // URL to get source code for building the image
+  source: `${ociPrefix}.source`,
+  // Source control revision identifier for the packaged software
+  revision: `${ociPrefix}.revision`
+};
+
+export const getImageSource = (annotation: Annotation) => {
+  const url = annotation?.[ociAnnotationKeys.source];
+  const revision = annotation?.[ociAnnotationKeys.revision];
+
+  if (!revision) {
+    return url;
+  }
+
+  let baseUrl;
+
+  if (url.includes('github.com')) {
+    baseUrl = url
+      .replace(/^git@github.com:/, 'https://github.com/')
+      .replace(/^https?:\/\/github.com\//, 'https://github.com/')
+      .replace(/\.git$/, '');
+    return `${baseUrl}/commit/${revision}`;
+  } else if (url.includes('gitlab.com')) {
+    baseUrl = url
+      .replace(/^git@gitlab.com:/, 'https://gitlab.com/')
+      .replace(/^https?:\/\/gitlab.com\//, 'https://gitlab.com/')
+      .replace(/\.git$/, '');
+    return `${baseUrl}/-/commit/${revision}`;
+  } else if (url.includes('bitbucket.org')) {
+    baseUrl = url
+      .replace(/^git@bitbucket.org:/, 'https://bitbucket.org/')
+      .replace(/^https?:\/\/bitbucket.org\//, 'https://bitbucket.org/')
+      .replace(/\.git$/, '');
+    return `${baseUrl}/commits/${revision}`;
+  }
+
+  return url;
+};
```

Comment:
Isn't better to use something similar to this? https://github.com/argoproj/argo-cd/blob/master/ui/src/app/shared/components/urls.ts

Reply:
I think its fine